controllers/tempo/tempostack_controller_test.go

```diff
@@ -419,3 +421,74 @@ func TestTLSEnable(t *testing.T) {
 		assert.Contains(t, names, fmt.Sprintf("tempo-%s-ca-bundle", nsn.Name))
 	}
 }
+
+func TestPruneIngress(t *testing.T) {
+	// Create object storage secret and Tempo CR
+	nsn := types.NamespacedName{Name: "prune-ingress-test", Namespace: "default"}
+	storageSecret := createSecret(t, nsn)
+	tempo := &v1alpha1.TempoStack{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nsn.Name,
+			Namespace: nsn.Namespace,
+		},
+		Spec: v1alpha1.TempoStackSpec{
+			Images: configv1alpha1.ImagesSpec{
+				Tempo:      "docker.io/grafana/tempo:1.5.0",
+				TempoQuery: "docker.io/grafana/tempo-query:1.5.0",
+			},
+			Storage: v1alpha1.ObjectStorageSpec{
+				Secret: storageSecret.Name,
+			},
+			Components: v1alpha1.TempoComponentsSpec{
+				QueryFrontend: v1alpha1.TempoQueryFrontendSpec{
+					JaegerQuery: v1alpha1.JaegerQuerySpec{
+						Enabled: true,
+						Ingress: v1alpha1.JaegerQueryIngressSpec{
+							Type: v1alpha1.IngressTypeIngress,
+						},
+					},
+				},
+			},
+		},
+	}
+	err := k8sClient.Create(context.Background(), tempo)
+	require.NoError(t, err)
+
+	// Reconcile
+	reconciler := TempoStackReconciler{
+		Client: k8sClient,
+		Scheme: testScheme,
+		FeatureGates: configv1alpha1.FeatureGates{
+			TLSProfile: string(configv1alpha1.TLSProfileIntermediateType),
+		},
+	}
+	req := ctrl.Request{
+		NamespacedName: nsn,
+	}
+	reconcileResult, err := reconciler.Reconcile(context.Background(), req)
+	require.NoError(t, err)
+	assert.Equal(t, false, reconcileResult.Requeue)
+
+	// Verify Ingress is created
+	ingressNsn := types.NamespacedName{Name: "tempo-prune-ingress-test-query-frontend", Namespace: "default"}
```

Comment:
Maybe, you should do the same for routes (just to increase the coverage)

Reply:
The tempostack_controller tests use a Kubernetes control plane (`KUBEBUILDER_ASSETS` env var in the Makefile).
When I create a unit test with a Route object I get a panic when calling `obj.GetName()` on the Route object.
I already expected that it won't work, but I'm surprised that it panics :smiling_face_with_tear:

Reply:
So afaics we can only test Routes in e2e tests, when the OpenShift router is installed.